### PR TITLE
added bdd for listing products 

### PR DESCRIPTION
**Pull request by vignesh3193**
BDD for listing products working
added a data_load function in server to send some data to redis
corrected spelling of initialize_redis() function

### CODE REVIEW

**Review by ephraimrosenfeld — CHANGES_REQUESTED** [review]
I believe the command to run the bdd tests should be added to the `.travis.yml` file.

under the `script` section of the file, add a new line:
```  - behave```

**Comment by ephraimrosenfeld** [comment]
@vignesh3193 , you need to add the dependencies to the `requirements.txt` file:
```
# BDD
behave==1.2.5
selenium==3.3.1
```
This is the reason the build failed on Travis CI.

File: features/products.feature [hunk 1]
```diff
@@ -12,22 +12,27 @@ Scenario: The server is running
...
 Scenario: List all products
-    When I visit "/products"
-    Then I should see "iPhone 7"
-    And I should see "ActiveCare Hair Dryer"
-    And I should see "Apple MacBook Pro"
+	Given the following products
+        | id | name                  | category             | discontinued | price |
+        |  1 | iPhone 7              | electronics          | False        | 800   |
+        |  2 | ActiveCare Hair Dryer | bathroom appliances  | False        | 200   |
+        |  2 | Apple MacBook Pro     | electronics          | False        | 2100  |
```
Comment (ephraimrosenfeld) [inline features/products.feature]: You have two items, (Hair Dryer and Macbook Pro) with the same id (2)

**Review by ephraimrosenfeld — CHANGES_REQUESTED** [review]

**Comment by ephraimrosenfeld** [comment]
When I run the `behave` test locally (after including the dependencies in the `requirements.txt` file) I get some test failure issue that it cannot read the `price` attributes from the Feature file.

**Comment by vignesh3193** [comment]
The ID is given by the system so it doesn't really matter but I will change it. The tests were working correctly on my machine. I will take a look. Apologies about the requirements file, I'm really not sure how to use travis.

**Comment by ephraimrosenfeld** [comment]
I made changes to the `BDD` branch. What I believe was happening is that, on your local/dev environment you've set the `newkey` value in your db as a result of running the server web application - but this value never gets set when running TDD/BDD tests. This explains why the tests were failing on Travis - which only runs the unit and integration tests. The change is set this value in TDD/BDD tests as well.

**Pull request merged**